pkg/proto/configuration/global/global.proto

```diff
@@ -57,6 +57,18 @@ message PrometheusPushgatewayConfiguration {
   google.protobuf.Duration push_interval = 5;
 }
 
+// Parameters for a "token bucket" for deciding when to sample.
+message ConstantRateSamplingPolicy {
+  // Number of tokens to put in the bucket per second.
+  int32 tokens_per_second = 1;
+
+  // Maximum number of tokens that may be held in the bucket.
+  int32 max_tokens = 2;
+
+  // Number of tokens required to consume from the bucket to sample a trace.
+  int32 tokens_per_sample = 3;
```

Comment:
I guess this was added to support some form of fractions. Is that really a realistic use case? My intuition says that a schema that's more like this would have been adequate for most users:
```
message ConstantRateSamplingPolicy {
  int64 samples_per_period = 1;
  google.protobuf.Duration period = 2;
}
```

Reply:
Yes, for example the ability to do 1 sample every 2 or 10 seconds. Configuring the period would be an alternate way to accomplish that. Will switch to that. I definitely overcomplicated that.